frontend/src/components/PasswordReset/index.tsx

```diff
@@ -0,0 +1,198 @@
+import { setErrors } from "~/lib/form";
+import FormTitle from "./FormTitle";
+import styles from "./PasswordReset.module.css";
+
+type State = {
+  isAlreadySignedIn: boolean;
+  isPasswordUpdated: boolean;
+  isTokenValid: boolean;
+}
+
+type Form = {
+  passwordResetToken: string;
+  userPassword: string;
+  userPassword2: string;
+}
+
+export default function PasswordReset() {
+  const { showBoundary } = useErrorBoundary();
+  const search = useSearch();
+  const user = useUser();
+  const token = new URLSearchParams(search).get("t");
+  const [state, setState] = useState<State>({ isAlreadySignedIn: false, isPasswordUpdated: false, isTokenValid: false });
+  const { clearErrors, formState: { errors, isSubmitting }, handleSubmit, register, reset, setError } = useForm<Form>();
+
+  useEffect(() => {
+    if (user) {
+      setState((prev) => ({ ...prev, isAlreadySignedIn: true }));
+    }
+  }, [user]);
+
+  useEffect(() => {
+    setState((prev) => ({ ...prev, isTokenValid: !!token }));
+  }, [token]);
+
+  useEffect(() => {
+    if (!!errors.passwordResetToken) {
+      setState((prev) => ({ ...prev, isTokenValid: false }));
+
+      reset();
+    }
+  }, [errors.passwordResetToken]);
+
+  async function submit(data: Form) {
+    try {
+      const response = await api.post("/authentication/reset-user-password", {
+        ...data,
+      });
+
+      const responseJson = await response.json();
+
+      if (response.ok) {
+        setState((prev) => ({ ...prev, isPasswordUpdated: true }));
+      } else {
+        setErrors(setError, responseJson);
+      }
+    } catch (error) {
+      showBoundary(error);
+    }
+  }
+
+  return (
+    <>
+      <Title>Reset password</Title>
+
+      <Layout className={styles.layout}>
+        {state.isAlreadySignedIn ? (
+          <div className="container">
+            <FormTitle
+              actions={(
+                <Link className="button button-lg button-blue" to="/settings">
+                  Go to settings
+                </Link>
+              )}
+              description="You can reset your password from user settings."
+              summary="Password reset"
+              title="You are already in!"
+            />
+          </div>
+        ) : state.isTokenValid ?
+          state.isPasswordUpdated ? (
+            <div className="container">
+              <FormTitle
+                actions={(
+                  <Link className="button button-lg button-blue" to="/sign-in">
+                    Sign in to your account
+                  </Link>
+                )}
+                description="You succesfully reset your password."
+                showGoBack={false}
+                summary="Password reset"
+                title="You are all set!"
+              />
+            </div>
+          ) : (
+            <div className="container">
+              <FormTitle
+                description="Enter a new password to access your account."
+                maxWidth="30rem"
```

Comment:
If we are going to make `FormTitle` a separate component, let's go with 1-2 maxWidth options instead of making it a CSS property 💐